go/worker/storage/worker.go

```diff
@@ -142,46 +152,53 @@ func (w *Worker) Start() error {
 		return nil
 	}
 
-	// Wait for all runtimes to terminate.
 	go func() {
 		defer close(w.quitCh)
-
-		for _, r := range w.runtimes {
-			<-r.Quit()
-		}
+		_ = w.Serve() // error logged as part of Serve already.
```

Comment:
You should move everything from `Start` and `Stop` to `Serve`, so that we can latter replace the service manager with the one that accepts `Service`s.